src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/ProjectModifier.cs

```diff
@@ -0,0 +1,209 @@
+                                {
+                                    modifiedClassDeclarationSyntax = documentBuilder.AddCodeSnippets(file, modifiedClassDeclarationSyntax); ;
+                                }
+                                //replace class node with all the updates.
+                                documentEditor.ReplaceNode(classDeclarationSyntax, modifiedClassDeclarationSyntax);
+                            }
+                            await documentBuilder.WriteToClassFile(fileName, filePath);
+                        }
+                    }
+                }
+            }
+        }
+
+        private CodeModifierConfig? GetCodeModifierConfig()
+        {
+            List<CodeModifierConfig> codeModifierConfigs = new List<CodeModifierConfig>();
+            if (!string.IsNullOrEmpty(_toolOptions.ProjectType))
+            {
+                var properties = typeof(Resources).GetProperties(BindingFlags.Static | BindingFlags.NonPublic)
+               .Where(p => p.PropertyType == typeof(byte[]))
+               .ToArray();
+
+                foreach (PropertyInfo propertyInfo in properties)
+                {
+                    if (propertyInfo.Name.StartsWith("cm") && propertyInfo.Name.Contains(_toolOptions.ProjectType))
+                    {
+                        byte[] content = (propertyInfo.GetValue(null) as byte[])!;
+                        CodeModifierConfig? projectDescription = ReadCodeModifierConfigFromFileContent(content);
+
+                        if (projectDescription == null)
+                        {
+                            throw new FormatException($"Resource file { propertyInfo.Name } could not be parsed. ");
+                        }
+                        codeModifierConfigs.Add(projectDescription);
+                    }
+                }
+            }
+
+            var codeModifierConfig = codeModifierConfigs
+                .Where(x => x.Identifier != null &&
+                       x.Identifier.Equals(_toolOptions.ProjectTypeIdentifier, StringComparison.OrdinalIgnoreCase))
+                        .FirstOrDefault();
+
+            // CodeModifierConfig, .csproj path cannot be null
+            if (codeModifierConfig != null &&
+                codeModifierConfig.Files != null &&
+                codeModifierConfig.Files.Any())
+            {
+                return codeModifierConfig;
+            }
+            return null;
+        }
+
+        private CodeModifierConfig? ReadCodeModifierConfigFromFileContent(byte[] fileContent)
+        {
+            string jsonText = Encoding.UTF8.GetString(fileContent);
+            return JsonSerializer.Deserialize<CodeModifierConfig>(jsonText);
+        }
+
+        internal string GetClassName(string? className)
+        {
+            string formattedClassName = string.Empty;
+            if (!string.IsNullOrEmpty(className))
+            {
+                string[] blocks = className.Split(".cs");
```

Comment:
does it work with Namespace.ClassName.cs?

Reply:
@abpiskunov  Oh I've never seen a class file with Namespace.ClassName.cs. Since I am controlling which classes to make code modifications to (mostly default template files), don't think I need this edge case covered yet. Will change if needed.

Reply:
i would still do it - users can change your template but you would fail here after that. Do it in on of next PRs please.